leaflet-treering.js

```diff
@@ -2743,6 +2806,173 @@ function ImageAdjustment(Lt) {
+      <div><p style="text-align:center;font-size:16px">&#9831; &#9831; &#9831; <button type="button" id="confirm-button" class="preferences-button"> Save & close </button> &#9831; &#9831; &#9831;</p></div>').addTo(Lt.viewer);
+  };
+
+  /**
+  * Based on initial data, selects buttons/dialog text
+  * @function selectedBtns
+  */
+  MeasurementOptions.prototype.selectedBtns = function () {
+    if (this.forwardDirection == true) {
+      document.getElementById("forward_radio").checked = true;
+    } else {
+      document.getElementById("backward_radio").checked = true;
+    };
+
+    if (this.subAnnual == true) {
+      document.getElementById("subannual_radio").checked = true;
+    } else {
+      document.getElementById("annual_radio").checked = true;
+    };
+  };
+
+  /**
+  * Changes direction & increment object to be saved
+  * @function prefBtnListener
+  */
+  MeasurementOptions.prototype.prefBtnListener = function () {
+    document.getElementById("forward_radio").addEventListener('change', (event) => {
+      if (event.target.checked == true) {
+        this.forwardDirection = true;
+      };
+    });
+
+    document.getElementById("backward_radio").addEventListener('change', (event) => {
+      if (event.target.checked == true) {
+        this.forwardDirection = false;
+      };
+    });
+
+    document.getElementById("annual_radio").addEventListener('change', (event) => {
+      if (event.target.checked == true) {
+        this.subAnnual = false;
+      };
+    });
+
+    document.getElementById("subannual_radio").addEventListener('change', (event) => {
+      if (event.target.checked == true) {
+        this.subAnnual = true
+      };
+    });
+  };
+
+  /**
+  * Open measurement options dialog
+  * @function enable
+  */
+  MeasurementOptions.prototype.enable = function() {
+    if (!this.dialog) {
+      this.dialog = this.displayDialog();
+    };
+
+    this.selectedBtns();
+
+    var forwardRadio = document.getElementById("forward_radio");
+    var backwardRadio = document.getElementById("backward_radio");
+    var annualRadio = document.getElementById("annual_radio");
+    var subAnnualRadio = document.getElementById("subannual_radio");
+    if ((Lt.data.points.length === 0 || !Lt.data.points[0]) && window.name.includes('popout')) {
+      forwardRadio.disabled = false;
+      backwardRadio.disabled = false;
+      annualRadio.disabled = false;
+      subAnnualRadio.disabled = false;
+      this.prefBtnListener();
+    } else { // lets users see preferences without being able to change them mid-measurement
+      forwardRadio.disabled = true;
+      backwardRadio.disabled = true;
+      annualRadio.disabled = true;
+      subAnnualRadio.disabled = true;
+    };
+
+    this.dialog.lock();
+    this.dialog.open();
+    this.btn.state('active');
+
+    $("#confirm-button").click(() => {
+      this.choice = true;
```

Comment:
this.choice is confusing - what's it there for? Can it be named more clearly?

Reply:
this.choice is for asset's not marked up before. When this.choice is false and a user goes to start measuring (clicks add points), the dialog box will trigger and disable 'create points'. Once the user selects their preferences (their choice), this.choice = true and the dialog box won't trigger the next time 'add point' is enabled. Maybe this.preferencesSelected?

Reply:
yeah, some name that makes it super voice

preferencesHaveBeenSet or something.

That way when you read the line that actually uses it, it's really clear what state the code is in